File: pdu/python/geometry_msgs/pdu_cdr_conv_Vector3.py
from .pdu_pytype_Vector3 import Vector3
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Vector3(writer: CdrWriter, src: Vector3):
    writer.write_float64(src.x)
    writer.write_float64(src.y)
    writer.write_float64(src.z)


def cdr_body_to_py_Vector3(reader: CdrReader, dst: Vector3):
    dst.x = reader.read_float64()
    dst.y = reader.read_float64()
    dst.z = reader.read_float64()
    return dst


def py_to_cdr_Vector3(src: Vector3) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Vector3(writer, src)
    return writer.bytes()


def cdr_to_py_Vector3(cdr_payload) -> Vector3:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Vector3()
    return cdr_body_to_py_Vector3(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_Vector3.js
import { Vector3 } from './pdu_jstype_Vector3.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduVector3Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Vector3 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_float64(src.x);
        writer.write_float64(src.y);
        writer.write_float64(src.z);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Vector3 } dst
     * @returns { Vector3 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.x = reader.read_float64();
        dst.y = reader.read_float64();
        dst.z = reader.read_float64();
        return dst;
    }

    /**
     * @param { Vector3 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Vector3 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Vector3());
    }
}
